nemo/collections/tts/modules/submodules.py

```diff
@@ -709,18 +709,29 @@ class SpeakerEncoder represents speakers representation.
     This module can combine GST (global style token) based speaker embeddings and lookup table speaker embeddings.
     """
 
-    def __init__(self, lookup_module=None, gst_module=None):
+    def __init__(self, lookup_module=None, gst_module=None, precompute=False, precompute_embedding_dim=None):
         """
         lookup_module: Torch module to get lookup based speaker embedding
         gst_module: Neural module to get GST based speaker embedding
+        precompute: Use precompute speaker embedding
         """
         super(SpeakerEncoder, self).__init__()
+
+        # Multi-speaker embedding
         self.lookup_module = lookup_module
+
+        # Reference speaker embedding
         self.gst_module = gst_module
 
+        if precompute:
```

Comment:
Can we have the check be if `precomputed_embedding_dim` is not None, instead of a separate `precompute` flag?

Reply:
If user want to use precompute embedding, I think setting a bool value is better than setting a dimension value?

Reply:
Having 2 parameters that are coupled but optional is awkward. If you keep it as is there should be an assert which validates that the dimension is None when the flag is false, or dimension is not None when the flag is true (similar to the XOR in https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/tts/data/dataset.py#L443-L445).

Normally one would provide dependent arguments as a single module or as a config object to ensure user provides all or none of them, but in this case there is only 1 argument to configure so checking if its None is also fine.

Reply:
Use only one parameter `precomputed_embedding_dim`